### PR TITLE
L1 TK hybrid code

### PR DESCRIPTION
**Pull request by skinnari**
Update to L1 tracking code, switching to the hybrid emulation code.

### CODE REVIEW

**Comment by skinnari** [comment]
the additional commit here is a fix to improve the performance of displaced (extended) tracks

**Comment by skinnari** [comment]
comparison of performance for ttbar PU=200 is available here: 

https://cernbox.cern.ch/index.php/s/9Qz7zsIhs6NYyIK

as shown, the performance is improved with the hybrid algorithm.

**Comment by skinnari** [comment]
@rekovic will you integrate this?

**Comment by rekovic** [comment]
@skinnari This does not compile.  Please fix.
Attn @cerminar

```
In file included from /afs/cern.ch/work/r/rekovic/private/relaeses/10x/CMSSW_10_6_1_patch2-Phase2-test/src/L1Trigger/TrackFindingTracklet/interface/Sector.h:32:0,
                 from /afs/cern.ch/work/r/rekovic/private/relaeses/10x/CMSSW_10_6_1_patch2-Phase2-test/src/L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc:87:
/afs/cern.ch/work/r/rekovic/private/relaeses/10x/CMSSW_10_6_1_patch2-Phase2-test/src/L1Trigger/TrackFindingTracklet/interface/FitTrack.h:13:10: fatal error: L1Trigger/TrackFindingTMTT/interface/L1track3D.h: No such file or directory
 #include "L1Trigger/TrackFindingTMTT/interface/L1track3D.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
In file included from /afs/cern.ch/work/r/rekovic/private/relaeses/10x/CMSSW_10_6_1_patch2-Phase2-test/src/L1Trigger/TrackFindingTracklet/interface/Sector.h:32:0,
                 from /afs/cern.ch/work/r/rekovic/private/relaeses/10x/CMSSW_10_6_1_patch2-Phase2-test/src/L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc:87:
/afs/cern.ch/work/r/rekovic/private/relaeses/10x/CMSSW_10_6_1_patch2-Phase2-test/src/L1Trigger/TrackFindingTracklet/interface/FitTrack.h:13:10: fatal error: L1Trigger/TrackFindingTMTT/interface/L1track3D.h: No such file or directory
 #include "L1Trigger/TrackFindingTMTT/interface/L1track3D.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

**Comment by skinnari** [comment]
@rekovic I am very confused. How are you compiling this? The error is that the file called "L1Trigger/TrackFindingTMTT/interface/L1track3D.h" does not exist, but the file is added in this commit?

File: L1Trigger/TrackFindingTMTT/interface/L1track3D.h [hunk 1]
```diff
@@ -0,0 +1,245 @@
+#ifndef __L1track3D_H__
```
Comment (skinnari) [inline L1Trigger/TrackFindingTMTT/interface/L1track3D.h:1]: @rekovic this is the file -- it is there, and code compiled just fine for me.

**Pull request merged**